ui/app/src/model/datasource-api.ts

```diff
@@ -12,40 +12,42 @@
 // limitations under the License.
 
 import { Datasource, GlobalDatasource } from '@perses-dev/core';
-import { DatasourceStoreProviderProps } from '@perses-dev/dashboards';
+import { DatasourceApi } from '@perses-dev/dashboards';
+import { fetchDatasourceList, fetchGlobalDatasourceList } from './datasource-client';
 
-export function useDatasourceApi(): DatasourceStoreProviderProps['datasourceApi'] {
+export function useDatasourceApi(): DatasourceApi {
   return {
-    getDatasource: async (/*project, selector*/) => {
-      // TODO: Convert selector to appropriate request params and fetchJson to get it from backend using project
-      // argument passed in
-      return undefined;
+    getDatasource: async (project, selector) => {
+      return fetchDatasourceList(project, selector.kind, selector.name ? undefined : true, selector.name).then(
+        (list) => {
+          // hopefully it should return at most one element even
+          if (list.length > 0 && list[0] !== undefined) {
```

Comment:
Nit: Since you're already checking `list[0] !== undefined`, you don't also need the `.length` check here.